assets/js/modules/adsense/components/settings/AdBlockingRecoveryApp/SetupMain.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback, useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import Layout from '../../../../../components/layout/Layout';
+import { Cell, Grid, Row } from '../../../../../material-components';
+import Content from './Content';
+import PlaceTagsStep from './steps/PlaceTagsStep';
+import CreateMessageStep from './steps/CreateMessageStep';
+import Link from '../../../../../components/Link';
+import Stepper from '../../../../../components/Stepper';
+import Step from '../../../../../components/Stepper/Step';
+import PageHeader from '../../../../../components/PageHeader';
+import { CORE_SITE } from '../../../../../googlesitekit/datastore/site/constants';
+import {
+	AD_BLOCKING_RECOVERY_SETUP_CREATE_MESSAGE_CTA_CLICKED,
+	ENUM_AD_BLOCKING_RECOVERY_SETUP_STATUS,
+	ENUM_AD_BLOCKING_RECOVERY_SETUP_STEP,
+	MODULES_ADSENSE,
+} from '../../../datastore/constants';
+import { CORE_UI } from '../../../../../googlesitekit/datastore/ui/constants';
+import { CORE_LOCATION } from '../../../../../googlesitekit/datastore/location/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function SetupMain() {
+	const settingsURL = useSelect(
+		( select ) =>
+			`${ select( CORE_SITE ).getAdminURL(
+				'googlesitekit-settings'
+			) }#/connected-services/adsense`
+	);
+	const createMessageCTAClicked = useSelect(
+		( select ) =>
+			!! select( CORE_UI ).getValue(
+				AD_BLOCKING_RECOVERY_SETUP_CREATE_MESSAGE_CTA_CLICKED
+			)
+	);
+	const dashboardURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getAdminURL( 'googlesitekit-dashboard' )
+	);
+	const initialActiveStep = useSelect( ( select ) => {
+		const adBlockingRecoverySetupStatus =
+			select( MODULES_ADSENSE ).getAdBlockingRecoverySetupStatus();
+
+		switch ( adBlockingRecoverySetupStatus ) {
+			case '':
+				return ENUM_AD_BLOCKING_RECOVERY_SETUP_STEP.PLACE_TAGS;
+			case ENUM_AD_BLOCKING_RECOVERY_SETUP_STATUS.TAG_PLACED:
+				return ENUM_AD_BLOCKING_RECOVERY_SETUP_STEP.CREATE_MESSAGE;
+			case ENUM_AD_BLOCKING_RECOVERY_SETUP_STATUS.SETUP_CONFIRMED:
+				return ENUM_AD_BLOCKING_RECOVERY_SETUP_STEP.COMPLETE;
+		}
```

Comment:
This mapping could be defined once at the top and then this could be simplified to `statusStepMap[status] || ENUM_AD_BLOCKING_RECOVERY_SETUP_STEP.PLACE_TAGS`

Not a big deal but it is a little hard to read as-is.

Reply:
Very good idea, updated, let me know if this looks better now, thank you!